### PR TITLE
Add react-social to social bar component [#140858491]

### PR DESCRIPTION
**Pull request by naomiyaki**
[Delivers #140858491]

### CODE REVIEW

**Comment by zdavis** [comment]
FYI: I rebased this onto development to fix CI tests.

File: client/src/components/frontend/Resource/Detail.js [hunk 1]
```diff
@@ -10,12 +10,14 @@ export default class ResourceDetail extends Component {
   static propTypes = {
     projectId: PropTypes.string,
     projectUrl: PropTypes.string,
+    resourceUrl: PropTypes.string,
```
Comment (zdavis) [inline client/src/components/frontend/Resource/Detail.js]: It's OK to not have this resourceUrl be required, but if it's not, then we shouldn't render the ShareBar if it's absent or blank. Or, alternately, we can make it required and always render the ShareBar.

File: client/src/components/frontend/ResourceCollection/Detail.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ export default class ResourceCollectionDetail extends PureComponent {
   static propTypes = {
     collection: PropTypes.object,
     project: PropTypes.object,
+    collectionUrl: PropTypes.string,
```
Comment (zdavis) [inline client/src/components/frontend/ResourceCollection/Detail.js]: Same comment as above—the collectionUrl should be required, or the share rendering should be conditional, I think.

File: client/src/components/frontend/Utility/ShareBar.js [hunk 3]
```diff
@@ -1,34 +1,78 @@
...
+    });
+  }
+
   render() {
+    if (!this.state.inBrowser) return null;
+
+    const twitterWindowOptions = ["", "", "width=600,height=300"];
+    const twitterAppId = this.props.twitterAppId ? this.props.twitterAppId : null;
+    const facebookAppId = this.props.facebookAppId ? this.props.facebookAppId : null;
+    {/* Todo: The hostname needs to be inserted here */}
```
Comment (zdavis) [inline client/src/components/frontend/Utility/ShareBar.js]: Is this todo still necessary? It's probably fine to get it from window.location, right?

File: client/src/containers/frontend/CollectionDetail.js [hunk 4]
```diff
@@ -107,7 +107,11 @@ class CollectionDetailContainer extends PureComponent {
   render() {
     const project = this.props.project;
     const collection = this.props.collection;
-    if (!project && !collection) return null;
+
+    if (!project || !collection) return null;
```
Comment (zdavis) [inline client/src/containers/frontend/CollectionDetail.js]: Why the condition change? Isn't it right to return null if we're missing the project _or_ the collection?

**Comment by zdavis** [comment]
This might be fine as is — just look through my comments and let me know if you think they warrant changes.

File: client/src/components/frontend/Utility/ShareBar.js (hunk 3 above)
Comment (naomiyaki) [inline client/src/components/frontend/Utility/ShareBar.js]: Nope! Done did.

File: client/src/containers/frontend/CollectionDetail.js (hunk 4 above)
Comment (zdavis) [inline client/src/containers/frontend/CollectionDetail.js]: image

**Pull request merged**